OSSM-11032 [DOCS] Modify upgrade procedure for ambient mode (#101283)
openshift/openshift-docs: modules/ossm-updating-waypoint-proxies-with-inplace-strategy.adoc

// Module included in the following assemblies:
// update/ossm-updating-openshift-service-mesh-in-ambient-mode.adoc

:_mod-docs-content-type: PROCEDURE
[id="ossm-updating-waypoint-proxies-with-inplace-strategy_{context}"]
= Updating waypoint proxies with InPlace strategy in ambient mode

[role="_abstract"]

> In ambient mode, you can upgrade waypoint proxies to the latest control plane version without restarting application workloads because they are deployed and managed as separate Gateway API resources that scale and upgrade independently.

Reviewer: ```suggestion
During an `InPlace` update in ambient mode, waypoint proxies are going to be updated to the latest control plane version without restarting application workloads because they are deployed and managed as separate Gateway API resources that scale and upgrade independently.
```

.Prerequisites

* You have updated the {istio} control plane with `InPlace` update strategy.

.Procedure

> . Verify that the waypoint proxy pods are running with the updated control plane by entering the following command:

Reviewer: ```suggestion
. Verify the waypoint proxy pods running in the application namespace by entering the following command:
```

+
[source,terminal]
----
$ oc get pods -n bookinfo -l gateway.networking.k8s.io/gateway-name=waypoint
----
+

> You should see an output similar to the following example:

Reviewer 2: Ohh.. I thought you were the author of the developer doc. @rafaelvzago, can you take a look.

> You should see an output similar to the following example:

Reviewer 3: @sridhargaddam You're right - `kubectl get pods` only shows if pods are running, not which control plane they're connected to.

Use `istioctl proxy-status` instead:

```bash
$ istioctl proxy-status | grep waypoint
waypoint-5d9c8b7f9-abc12.bookinfo     SYNCED     SYNCED     SYNCED     SYNCED     istiod-1-26-4-7b9f8c5d6-xyz78.istio-system     1.26.4
```

This shows the actual control plane connection, sync status, and version. The revision-specific istiod name (e.g., `istiod-1-26-4-...`) confirms it's using the new revision.

Documented in https://github.com/istio-ecosystem/sail-operator/pull/1279

> You should see an output similar to the following example:

Author: Okay! Removing this step. The step that Rafael mentioned is already drafted as the next step.

+
[source,terminal]
----
NAME                       READY   STATUS    RESTARTS   AGE
waypoint-5d9c8b7f9-abc12   1/1     Running   0          5m
----

> . Confirm that the waypoint proxy is connected to the new control plane by running the following command:

Reviewer: ```suggestion
. Confirm that the waypoint proxy was updated proxy version by running the following command:
```
